### PR TITLE
[NON-MODULAR] No Tendril Chasms!

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request
Having your character literally qdel'd after a lagspike at the wrong time sucks, at least with lava, you have a chance to get out before you keel over!

That, and it's been constantly discussed and agreed on in discord that TGChasms just don't work with Skyrat's general design, due to "instant round removal<sup>:tm:</sup>", so!

Also, lavaland is a planet of... lava? Wouldn't lava rush in when the tendril pulls back in?

## How This Contributes To The Skyrat Roleplay Experience
Less round removals keep players able to play and drastically lightens admin workload should someone die in lava due to lag compared to chasms!

## Changelog

:cl:
qol: Destroyed tendrils spawn lava instead of chasms.
/:cl:

### CODE REVIEW

File: code/game/objects/structures/lavaland/necropolis_tendril.dm [hunk 1]
```diff
@@ -96,5 +96,10 @@ GLOBAL_LIST_INIT(tendrils, list())
 	visible_message(span_boldannounce("The tendril falls inward, the ground around it widening into a yawning chasm!"))
 	for(var/turf/T in RANGE_TURFS(2,src))
 		if(!T.density)
+			// SKYRAT EDIT - No chasms
+			/* Original code
 			T.TerraformTurf(/turf/open/chasm/lavaland, /turf/open/chasm/lavaland, flags = CHANGETURF_INHERIT_AIR)
+			*/
+			T.TerraformTurf(/turf/open/smooth, /turf/open/lava/smooth, flags = CHANGETURF_INHERIT_AIR)
```
Comment (tf-4) [inline code/game/objects/structures/lavaland/necropolis_tendril.dm]:
```suggestion
			T.TerraformTurf(/turf/open/lava/smooth/lava_land_surface, flags = CHANGETURF_INHERIT_AIR)
```

Comment (RimiNosha) [reply code/game/objects/structures/lavaland/necropolis_tendril.dm]: How the heck did lava go missing- anyways, thanks!

**Comment by HoodMilk** [comment]
We had this on a server I adminned for and it worked really well. It also helps during ash storms because you can clearly see lava compared to the chasms which are nearly invisible during storms.

**Comment by Iamgoofball** [comment]
miners literally have a tool to dodge the chasm
if you're not a miner and don't have access to it, why are you fighting tendrils?

**Comment by nevimer** [comment]
I will be shocked if this gets through, the last time I tried to bring this up, people (&maintainers) resisted so hard and said paraphrased "just git gud"...

...Then decided to make the jaunter work from your bags aswell instead of just your belt slot. Acknowledging that there is a problem, but decided to pussyfoot around the issue instead of just remove the issue. 

Wild. The ultimate move would be to just make chasms not exist because, again, skyrat isn't a battle royale or some shit and that mentality in game design needs to go.

**Comment by nikita-chekistov** [comment]
That's an interesting idea, but should it really be on the server? Lavaland is supposed to be deadly and unforgiving.
If a miner destroys a tendrill, he should understand, that he creates a trap for himself in the future.
And yes, we have jaunters to avoid this. If you cant afford it - you're probably not a miner and what are you even doing on lavaland.
Ashstorms also are supposed to be a pain in the ass. If you can't see the chasms during it, dont run in the fields while there is a storm.

What's next? Remove megafauna's ability to gib/dust?

**Comment by hugmeorelse** [comment]
It's not about getting good. It's about not lagging into a chasm. **The best move is to have people fall down a z-level if they fall into a chasm like in oldbase.**

**Comment by Zenitheevee** [comment]
> That's an interesting idea, but should it really be on the server? Lavaland is supposed to be deadly and unforgiving. If a miner destroys a tendrill, he should understand, that he creates a trap for himself in the future. And yes, we have jaunters to avoid this. If you cant afford it - you're probably not a miner and what are you even doing on lavaland. Ashstorms also are supposed to be a pain in the ass. If you can't see the chasms during it, dont run in the fields while there is a storm.
> 
> What's next? Remove megafauna's ability to gib/dust?

in a perfect world where SS13 can be ran stably without lag or hickups, I'd agree. 
But until then i'd say this is probably one of the better answers to the issue. Not the _best_, but sadly the best would make the server run horridly.

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
Instead of this, can you make chasms just spit you out somewhere random on the Z level they are on

**Comment by YakumoChen** [comment]
if this changes it won't qdel the chest if you don't snag it in time, making it no longer a timed event to grab the loot and run.

Keep it chasms and make chasms do something else. It should also be made so that if a crate falls in a chasm it's deleted no matter what.

**Comment by Gandalf2k15** [comment]
reopen this when you've made the requested changes

**Pull request closed without merging**